pytket/extensions/quantinuum/backends/leakage_gadget.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020-2023 Cambridge Quantum Computing
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Methods for generating a leakage detection Pytket Circuit."""
+
+
+from pytket import Circuit, Qubit, Bit, OpType
+from pytket.backends.backendresult import BackendResult # type: ignore
+from pytket.utils.outcomearray import OutcomeArray  # type: ignore
+from typing import List, Dict, Tuple, Counter, cast, Sequence
+
+def get_leakage_gadget_circuit(
+    circuit_qubit: Qubit, postselection_qubit: Qubit, postselection_bit: Bit
+) -> Circuit:
+    """
+    Returns a two qubit Circuit for detecting leakage errors.
+
+    :param circuit_qubit: Generated circuit detects whether leakage errors
+    have occurred in this qubit.
+    :type circuit_qubit: Qubit
+    :param postselection_qubit: Measured qubit to detect leakage error.
+    :type postselection_qubit: Qubit
+    :param postselection_bit: Leakage detection result is written to this bit.
+    :type postselection_bit: Bit
+    :return: Circuit for detecting leakage errors for specified ids.
+    :rtype: Circuit
+    """
+    c = Circuit()
+    c.add_qubit(circuit_qubit)
+    c.add_qubit(postselection_qubit)
+    c.add_bit(postselection_bit)
+    c.X(postselection_qubit)
+    c.add_barrier([circuit_qubit, postselection_qubit])
+    c.H(postselection_qubit).ZZMax(postselection_qubit, circuit_qubit)
```

Comment:
This seems to contain gates (like `H`) that are not normally in the backend-supported gateset. Should it be a (underscored) method of `QuantinuumBackend` that compiles this circuit to the target gateset, rather than a standalone method that doesn't know about the gateset?

Reply:
I think we should avoid hard-coding the gateset if possible, instead rebasing to the gate set reported (which is somewhat variable, or at least has been). I'm not sure if we need to be careful about compilation: I've been sternly warned in the past by error-correction people that compilation changes the error-corrective properties. But that may not apply here.

Reply:
Ok great. In which case done